libs/designsystem/helpers/src/content-mutation-observer.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, ElementRef, Input, OnDestroy } from '@angular/core';
+import { createComponentFactory, Spectator } from '@ngneat/spectator';
+import { TestHelper } from '@kirbydesign/designsystem/testing';
+import { observeContent } from './content-mutation-observer';
+
+@Component({
+  // eslint-disable-next-line @angular-eslint/component-selector
+  selector: 'content-observer-host',
+  template: '<div [innerHtml]="content"></div>',
+})
+class ContentObserverHostComponent implements OnDestroy {
+  @Input() content: string = 'Initial content';
+
+  private mutationObserver: MutationObserver;
+
+  constructor(private elementRef: ElementRef) {
+    const contentHost = this.elementRef.nativeElement;
+    this.mutationObserver = observeContent(contentHost, () => this.onContentChange());
+  }
+
+  public onContentChange() {}
+
+  ngOnDestroy(): void {
+    this.mutationObserver.disconnect();
+  }
+}
+
+describe('observeContent', () => {
+  let spectator: Spectator<ContentObserverHostComponent>;
+
+  const createComponent = createComponentFactory({
+    component: ContentObserverHostComponent,
+  });
+
+  beforeEach(async () => {
+    spectator = createComponent();
+
+    // Wait for MutationObserver to be ready, we dont want to trigger the callback on initialization
+    await TestHelper.waitForMutationObserver();
+  });
+
+  afterEach(() => {
+    spectator.fixture.destroy();
+  });
+
+  it('should create', () => {
+    expect(spectator.component).toBeTruthy();
+  });
+
+  it('should call callback when content changes', async () => {
+    const onContentChangeSpy = spyOn(spectator.component, 'onContentChange');
+
+    spectator.setInput('content', 'Updated content');
+    await TestHelper.waitForMutationObserver();
+    expect(onContentChangeSpy).toHaveBeenCalledTimes(1);
```

Comment:
Consider readability, Arrange -> Act -> Assert
```suggestion
    spectator.setInput('content', 'Updated content');
    await TestHelper.waitForMutationObserver();
    
    expect(onContentChangeSpy).toHaveBeenCalledTimes(1);
```